src/main/java/com/synopsys/integration/alert/workflow/startup/component/LastSearchDataMigration.java

```diff
@@ -0,0 +1,91 @@
+/**
+ * blackduck-alert
+ *
+ * Copyright (c) 2020 Synopsys, Inc.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.synopsys.integration.alert.workflow.startup.component;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Optional;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.core.annotation.Order;
+import org.springframework.stereotype.Component;
+
+import com.synopsys.integration.alert.common.enumeration.ConfigContextEnum;
+import com.synopsys.integration.alert.common.exception.AlertDatabaseConstraintException;
+import com.synopsys.integration.alert.common.persistence.accessor.ConfigurationAccessor;
+import com.synopsys.integration.alert.common.persistence.accessor.ProviderTaskPropertiesAccessor;
+import com.synopsys.integration.alert.common.persistence.model.ConfigurationModel;
+import com.synopsys.integration.alert.common.persistence.util.FilePersistenceUtil;
+import com.synopsys.integration.alert.provider.blackduck.BlackDuckProviderKey;
+import com.synopsys.integration.alert.provider.blackduck.tasks.BlackDuckAccumulator;
+
+@Component
+@Order(55)
+public class LastSearchDataMigration extends StartupComponent {
+    private static final Logger logger = LoggerFactory.getLogger(LastSearchDataMigration.class);
+    private static final String LAST_SEARCH_FILE = "blackduck-accumulator-task-last-search.txt";
+
+    // Because pg_stat_file requires admin privileges to run, we have to migrate the last search file data in code.
+    // We don't have admin privileges to postgres when alert starts.
+
+    private BlackDuckProviderKey providerKey;
```

Comment:
We can call this variable `blackDuckProviderKey` to make it more clear in the `initialize()` method that this is the only expected config in an upgrade.